packages/video_player/video_player_android/android/src/main/java/io/flutter/plugins/videoplayer/VideoPlayer.java

```diff
@@ -317,6 +351,74 @@ void sendInitialized() {
     }
   }
 
+
+  List<Messages.GetEmbeddedSubtitlesMessage> getEmbeddedSubtitles() {
+    List<Messages.GetEmbeddedSubtitlesMessage> subtitleItems = new ArrayList<>();
+    TrackGroupArray trackGroups;
+    int rendererIndex = 2;
+
+    MappingTrackSelector.MappedTrackInfo trackInfo = trackSelector.getCurrentMappedTrackInfo();
+    if (trackInfo == null) {
+      // TrackSelector not initialized
+      return subtitleItems;
+    }
+
+    trackGroups = trackInfo.getTrackGroups(rendererIndex);
+
+    for (int groupIndex = 0; groupIndex < trackGroups.length; groupIndex++) {
+      TrackGroup group = trackGroups.get(groupIndex);
+      for (int trackIndex = 0; trackIndex < group.length; trackIndex++) {
+
+        subtitleItems.add(
+                new Messages.GetEmbeddedSubtitlesMessage.Builder()
+                        .setLanguage(group.getFormat(trackIndex).language)
+                        .setLabel(group.getFormat(trackIndex).label)
+                        .setTrackIndex((long) trackIndex)
+                        .setGroupIndex((long) groupIndex)
+                        .setRenderIndex((long) rendererIndex)
+                        .build()
+        );
+      }
+    }
+
+    return subtitleItems;
+  }
+
+  void setEmbeddedSubtitles(Long trackIndex, Long groupIndex, Long rendererIndex) {
+    this.textTrackIndex = trackIndex;
+    if(trackIndex != null && groupIndex != null && rendererIndex != null) {
+        boolean isDisabled;
+        TrackGroupArray trackGroups;
+
+        MappingTrackSelector.MappedTrackInfo trackInfo =
+                trackSelector == null ? null : trackSelector.getCurrentMappedTrackInfo();
+        if (trackSelector == null || trackInfo == null) {
+            // TrackSelector not initialized
+            return;
+        }
+
+        trackGroups = trackInfo.getTrackGroups(Math.toIntExact(rendererIndex));
+        DefaultTrackSelector.Parameters parameters = trackSelector.getParameters();
+        isDisabled = parameters.getRendererDisabled(Math.toIntExact(rendererIndex));
+
+        DefaultTrackSelector.Parameters.Builder parametersBuilder = trackSelector.buildUponParameters()
+                .setRendererDisabled(C.TRACK_TYPE_VIDEO, false);
+        parametersBuilder.setRendererDisabled(Math.toIntExact(rendererIndex), isDisabled);
+
+        TrackSelectionOverride override = new TrackSelectionOverride(trackGroups.get(Math.toIntExact(groupIndex)), Math.toIntExact(trackIndex));
+
+        parametersBuilder.clearOverrides();
+        parametersBuilder.addOverride(override);
+
+        trackSelector.setParameters(parametersBuilder);
+    }else{
+      Map<String, Object> event = new HashMap<>();
+      event.put("event", "subtitle");
+      event.put("value", "");
+      eventSink.success(event);
```

Comment:
Why is an event not added to the `eventSink` when embedded subtitles were successfully set? I assume it has something to do with going through `onCues()`, but I just wish to understand this decision better.